repos/backend/libs/definitions/definitionsParser.js

```diff
@@ -1,106 +1,84 @@
 
-let defCache = {}
-const DEFINITION_REGEX = new RegExp(/(Given|When|Then|test)\(('|"|`|\/)(.*)('|"|`|\/),/, 'gm')
-
-const getDefinitionContent = definitionMatch => {
-  const content = definitionMatch.input.split(definitionMatch[0]).pop()
-  return `${definitionMatch[0]}${content.split(definitionMatch[1]).shift()}`
-}
-
 class DefinitionsParser {
 
-  constructor(){
-    this.resetDefinitions()
-  }
-
-  resetDefinitions = () => { 
-    // Holds the loaded defs
-    // Used to check if a definition was already loaded
-    this.definitions = {}
-    defCache = {}
+  clear = () => {
+    parkin.steps.clear()
   }
 
   getDefinitions = async filePath => {
-    if(defCache[filePath]) return defCache[filePath]
-
-    // Holds the loaded defs fileModels by file name
-    // This is what will be returned to the frontend
-    defCache[filePath] = []
-
-    const { fileModel, definitions } = await this.parseDefinition(filePath)
-
-    definitions.map(({ match, type, variant, content }) => {
-      if(!this.validateMatch(filePath, match, type)) return
-
-      const definition = this.definitions[match] || new Definition(match, type, variant, content)
-      !this.definitions[match] && (this.definitions[match] = definition)
-      
-      // Add a reference back to the parent
-      definition.parentUuid = fileModel.uuid
-
-      // Add the definition to the fileModels ast.definition array
-      fileModel.ast.definitions.push(definition)
-      
-      return definition
+    const { fileModel } = await this.parseDefinition(filePath)
+
+    // The definitions get auto-loaded into the parkin instance
+    // from the require call in the parseDefinition method below
+    const definitions = parkin.steps.list()
+
+    definitions.map(def => {
+      // If the file model contains the step def
+      // And it's a valid match string
+      // Then add the def to the fileModels ast.definitions array
+      fileModel.content.includes(def.match.toString()) &&
+        this.validateMatch(def.match, def.type) &&
+        fileModel.ast.definitions.push({
+          ...def,
+          // Add a reference back to the parent
+          location: filePath
+        })
     })
 
-    defCache[filePath] = fileModel
-
-    return defCache[filePath]
+    return fileModel
   }
 
   parseDefinition = (filePath) => {
     return new Promise((res, rej) => {
-      const definitions = []
+      // We still want to load the file content
+      // Even if the require call fails
+      // So wrap if in a try catch, and log the error if it happends
+      let response
+      try {
+
+        // Always clear out the node require cache
+        // This ensure we get a fresh file every time
+        // Otherwise changed files would not get reloaded
+        if(require.cache[filePath])
+          delete require.cache[filePath]
+
+        // Require the file, to auto-load the definitions into parkin
+        // Later we'll pull them from parkin
+        response = require(filePath)
```

Comment:
Might be a good case for a jsutils function at some point:
```js
const cleanRequire = filePath => {
  delete require.cache[filePath]
  return require(filePath)
}
```

Reply:
Yah that would be a good one. Although, it's not often I need to do something like this. The next time I need it, I'll probably forget the method exists in jsutils...lol